linkage-monitor/src/test/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitorTest.java

```diff
@@ -96,23 +115,17 @@ public void testBomSnapshot()
         .inOrder();
   }
 
-  private final SymbolProblem classNotFoundProblem =
-      new SymbolProblem(new ClassSymbol("java.lang.Integer"), ErrorType.CLASS_NOT_FOUND, null);
-  private final SymbolProblem methodNotFoundProblem =
-      new SymbolProblem(
-          new MethodSymbol(
-              "io.grpc.protobuf.ProtoUtils",
-              "marshaller",
-              "(Lcom/google/protobuf/Message;)Lio/grpc/MethodDescriptor$Marshaller;",
-              false),
-          ErrorType.SYMBOL_NOT_FOUND,
-          new ClassFile(ClassPathEntry.of("foo:b:1.0.0", "foo/b-1.0.0.jar"), "java.lang.Object"));
-
   @Test
   public void generateMessageForNewError() {
     Set<SymbolProblem> baselineProblems = ImmutableSet.of(classNotFoundProblem);
-    ClassPathEntry jarA = ClassPathEntry.of("foo:a:1.2.3", "foo/a-1.2.3.jar");
-    ClassPathEntry jarB = ClassPathEntry.of("foo:b:1.0.0", "foo/b-1.0.0.jar");
+    
+    Artifact artifactA = new DefaultArtifact("foo:a:1.2.3")
+        .setFile(new File("foo/a-1.2.3.jar"));
+    Artifact artifactB = new DefaultArtifact("foo:b:1.0.0")
+        .setFile(new File("foo/b-1.0.0.jar"));
+    
+    ClassPathEntry jarA = new ClassPathEntry(artifactA);
+    ClassPathEntry jarB = new ClassPathEntry(artifactB);
```

Comment:
Before: 2 lines with 2 variables
After: 6 lines with 4 variables

I see the before is better.

Reply:
Arguably we don't have the right constructors yet. However the model code seems to be OK with this, and it's nice to keep the tests as close to the model code as reasonable. But my main goal is to remove a method that's only needed for testing. That's always a good thing. 

Could this become a test utility? Perhaps. Or we could use test fixtures to avoid duplicating code for object creation.

Reply:
I agree that the removal of the test-only method is a good thing if the lost benefit is minimal.

> Could this become a test utility? Perhaps.

Would you implement this `of()` as a test utility?

I haven't thought about setting up test fixtures for the test cases. If you think that's even better, I'm interested in what it looks like.